components/org.wso2.carbon.siddhi.extensions.installer.core/src/main/java/org/wso2/carbon/siddhi/extensions/installer/core/execution/SiddhiAppExtensionUsageDetectorImpl.java

```diff
@@ -0,0 +1,205 @@
+     */
+    private Optional<ExtensionConfig> matchByUniqueAttributeValue(SiddhiAppExtensionUsage usage) {
+        Optional<Map.Entry<String, ExtensionConfig>> matchedExtensionEntry =
+            extensionConfigs.entrySet().stream()
+                .filter(extension -> {
+                    ExtensionIdentifierConfig extensionIdentifier = extension.getValue().getIdentifier();
+                    if (extensionIdentifier != null && extensionIdentifier.isUniqueAttributeValueValid()) {
+                        String uniqueAttribute = extensionIdentifier.getUniqueAttribute();
+                        String uniqueAttributeValue = extensionIdentifier.getUniqueAttributeValue();
+                        return (Objects.equals(usage.getProperties().get(TYPE_KEY), extensionIdentifier.getType()) &&
+                            Objects.equals(usage.getProperties().get(uniqueAttribute), uniqueAttributeValue));
+                    }
+                    return false;
+                })
+                .findFirst();
+        return matchedExtensionEntry.map(Map.Entry::getValue);
+    }
+
+    /**
+     * Returns an optional extension config object,
+     * when the given usage matches an extension by the regex of a unique attribute's value,
+     * as specified in the extension's {@link ExtensionIdentifierConfig}.
+     * Extensions of different variations, that share the same value for 'type' in Siddhi,
+     * but are identifiable based on the regex of a unique attribute's value fall under this category.
+     * Eg:
+     * - type='cdc' (common).
+     * - url='jdbc:mysql://...' for MySQL.
+     * - url='jdbc:postgresql://...' for Postgres.
+     *
+     * @param usage Extension usage in a Siddhi app.
+     * @return Optional Extension config object if found.
+     */
+    private Optional<ExtensionConfig> matchByUniqueAttributeValueRegex(SiddhiAppExtensionUsage usage) {
+        Optional<Map.Entry<String, ExtensionConfig>> matchedExtensionEntry =
+            extensionConfigs.entrySet().stream()
+                .filter(extension -> {
+                    ExtensionIdentifierConfig extensionIdentifier = extension.getValue().getIdentifier();
+                    if (extensionIdentifier != null && extensionIdentifier.isUniqueAttributeValueRegexValid()) {
+                        String uniqueAttribute = extensionIdentifier.getUniqueAttribute();
+                        String uniqueAttributeValueRegex = extensionIdentifier.getUniqueAttributeValueRegex();
+                        return (
+                            Objects.equals(usage.getProperties().get(TYPE_KEY), extensionIdentifier.getType()) &&
+                                usage.getProperties().get(uniqueAttribute) != null &&
+                                usage.getProperties().get(uniqueAttribute).matches(uniqueAttributeValueRegex));
+                    }
+                    return false;
+                })
+                .findFirst();
```

Comment:
shall we move this to 152

Reply:
Check other places as well

Reply:
Updated in https://github.com/wso2/carbon-analytics/pull/1840/commits/687d9d726cb4be51cef81b489166d694c78b91dd